metronome/checkpointing/service/src/io/iohk/metronome/checkpointing/service/CheckpointingService.scala

```diff
@@ -60,14 +68,12 @@ class CheckpointingService[F[_]: Sync: Applicative, N](
     updateLedger(block).flatMap {
       case Some(ledger) =>
         if (block.hash != commitQC.blockHash)
-          saveState(block.header, ledger).as(true)
+          false.pure[F]
```

Comment:
As you observed in #45 if we were to save this block as the last executed we could potentially lose the checkpoint information from preceding blocks. Alternatively, to facilitate persisting execution of each (non-committed) block, we could modify the ledger to also contain the checkpoint proof, or at least the containing header's hash.

WDYT?

Reply:
I think it's fine this way.